indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Encoding.kt

```diff
@@ -518,3 +510,48 @@ fun ByteArray.ensureSize(size: Int): ByteArray = (this.size - size).let { toDrop
 
 @Suppress("NOTHING_TO_INLINE")
 inline fun ByteArray.ensureSize(size: UInt) = ensureSize(size.toInt())
+
+/**
+ * Encodes this number using varint encoding as used within ASN.1: groups of seven bits are encoded into a byte,
+ * while the highest bit indicates if more bytes are to come
+ */
+fun ULong.toAsn1VarInt(): ByteArray {
+    if (this < 128u) return byteArrayOf(this.toByte()) //Fast case
+    var offset = 0
+    var result= mutableListOf<Byte>()
+
+    var b0 = (this shr offset and 0x7FuL).toByte()
+    while ((this shr offset > 0uL) || offset == 0) {
+        result += b0
+        offset += 7
+        if (offset > (ULong.SIZE_BITS - 1)) break //End of Fahnenstange
```

Comment:
this does not feel like it should be silently ignored; `check()`?

Reply:
what do you mean? this is just the terminating condition

Reply:
is it? i'd think the terminating condition is `this shr offset > 0uL`, and this is just "there are still bits but we have nowhere to put them", which sounds like an error condition to me